crmd/te_actions.c

```diff
@@ -53,9 +53,31 @@ te_start_action_timer(crm_graph_t * graph, crm_action_t * action)
 static gboolean
 te_pseudo_action(crm_graph_t * graph, crm_action_t * pseudo)
 {
+    const char *task = crm_element_value(pseudo->xml, XML_LRM_ATTR_TASK);
+
     /* Check action for Pacemaker Remote node side effects */
     remote_ra_process_pseudo(pseudo->xml);
 
+    /* send to peers as well? */
+    if (safe_str_eq(task, CRM_OP_MAINTENANCE_NODES)) {
+        GHashTableIter iter;
+        crm_node_t *node = NULL;
+
+        g_hash_table_iter_init(&iter, crm_peer_cache);
+        while (g_hash_table_iter_next(&iter, NULL, (gpointer *) &node)) {
+            xmlNode *cmd = NULL;
+
+            if (safe_str_eq(fsa_our_uname, node->uname)) {
+                continue;
+            }
+
+            cmd = create_request(task, pseudo->xml, node->uname,
+                                 CRM_SYSTEM_CRMD, CRM_SYSTEM_TENGINE, NULL);
+            send_cluster_message(node, crm_msg_crmd, cmd, FALSE);
+            free_xml(cmd);
+        }
+    }
+
```

Comment:
I think it will be simpler to pull the logic out of remote_ra_process_pseudo() into its own function to be called when receiving this message, so that here we can just send a single broadcast message, and the DC will process it upon receiving it just like the other cluster nodes.